test/integration/newapp_test.go

```diff
@@ -1328,6 +1351,95 @@ func TestBuildOutputCycleDetection(t *testing.T) {
 				return err.Error() == fmt.Errorf("%v, set a different tag with --to", e).Error()
 			},
 		},
+		{
+			name: "successful build with warning that output docker-image may trigger input ImageStream change",
+			config: &cmd.AppConfig{
+				GenerationInputs: cmd.GenerationInputs{
+					OutputDocker: true,
+					To:           "centos/ruby-22-centos7",
+					Dockerfile:   "FROM centos/ruby-22-centos7",
+				},
+				Resolvers: cmd.Resolvers{
+					DockerSearcher: app.DockerClientSearcher{
+						Client:           &apptest.FakeDockerClient{},
+						Insecure:         true,
+						RegistrySearcher: &ExactMatchDirectTagDockerSearcher{},
+					},
+				},
+			},
+			expected: map[string][]string{
+				"buildConfig": {"ruby-22-centos7"},
+				"imageStream": {"ruby-22-centos7"},
+			},
+			checkOutput: func(stdout, stderr io.Reader) error {
+				got, err := ioutil.ReadAll(stderr)
+				if err != nil {
+					return err
+				}
+				want := "--> WARNING: output image of \"default/centos/ruby-22-centos7\" must be different than input\n"
+				if string(got) != want {
+					return fmt.Errorf("stderr: got %q; want %q", got, want)
+				}
+				return nil
+			},
+		},
+		{
+			name: "successful build with warning that output docker-image may trigger input ImageStream change; latest variation",
+			config: &cmd.AppConfig{
+				GenerationInputs: cmd.GenerationInputs{
+					OutputDocker: true,
+					To:           "centos/ruby-22-centos7",
+					Dockerfile:   "FROM centos/ruby-22-centos7:latest",
+				},
+				Resolvers: cmd.Resolvers{
+					DockerSearcher: app.DockerClientSearcher{
+						Client:           &apptest.FakeDockerClient{},
+						Insecure:         true,
+						RegistrySearcher: &ExactMatchDirectTagDockerSearcher{},
+					},
+				},
+			},
+			expected: map[string][]string{
+				"buildConfig": {"ruby-22-centos7"},
+				"imageStream": {"ruby-22-centos7"},
+			},
+			checkOutput: func(stdout, stderr io.Reader) error {
+				got, err := ioutil.ReadAll(stderr)
+				if err != nil {
+					return err
+				}
+				want := "--> WARNING: output image of \"default/centos/ruby-22-centos7:latest\" must be different than input\n"
```

Comment:
can we change this actual warning message:
s/must.*/should be different from the input to avoid circular build triggering\n/

(since it's a warning, not an error, it doesn't make sense to say "must")